catalog/transformer/catalog-transformer-pptx/src/main/java/ddf/catalog/transformer/input/pptx/PptxInputTransformer.java

```diff
@@ -0,0 +1,210 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input.pptx;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.pdfbox.tools.imageio.ImageIOUtil;
+import org.apache.poi.sl.usermodel.SlideShow;
+import org.apache.poi.sl.usermodel.SlideShowFactory;
+import org.apache.poi.xslf.usermodel.XMLSlideShow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.FileBackedOutputStream;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+
+/**
+ * This is a decorator class that adds a thumbnail for PPTX files. It relies on
+ * an injected {@link InputTransformer} to extract the metadata, and then it generates a
+ * thumbnail image with Apache POI.
+ */
+public class PptxInputTransformer implements InputTransformer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PptxInputTransformer.class);
+
+    private static final int RESOLUTION_DPI = 44;
+
+    private static final float IMAGE_QUALITY = 1.0f;
+
+    private static final float IMAGE_HEIGHTWIDTH = 128;
+
+    private static final String FORMAT_NAME = "jpg";
+
+    private static final int FILE_BACKED_THRESHOLD = 1000000;
+
+    private final InputTransformer inputTransformer;
+
+    /**
+     * The inputTransformer parameter will be used to generate the basic metadata. If the parameter
+     * is null, then a {@link NullPointerException} will be thrown.
+     *
+     * @param inputTransformer must be non-null
+     * @throws NullPointerException
+     */
+    public PptxInputTransformer(InputTransformer inputTransformer) {
+
+        if (inputTransformer == null) {
+            throw new NullPointerException("The inputTransformer parameter must be non-null");
+        }
+
+        this.inputTransformer = inputTransformer;
+    }
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+
+        if (input == null) {
+            throw new CatalogTransformerException("Cannot transform null input.");
+        }
+
+        /*
+         * A FileBackedOutputStream is used because we need to consume the stream twice. Once
+         * for the injected inputTransformer and once for Apache POI.
+         */
+        try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(
```

Comment:
I understand your reasoning for providing an explanation, but self-documenting code is important. If this is absolutely necessary, pull out the offending logic into it's own private method, and add this JAVADOC to the top of that private method.